Add freethreading support

isuruf opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

matplotlib has wheels in latest release

isuruf committed:
Add freethreading support

isuruf commented:
@conda-forge-admin rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

tacaswell commented:
To be clear, this is adding the build options for freethreading, not claiming mpl is freethreading safe?

dopplershift commented:
What does "freethreading safe" mean?

ngoldbaum commented:
> To be clear, this is adding the build options for freethreading, not claiming mpl is freethreading safe?

This is building binaries that support the new CPython ABI under the free-threaded build. This is related but orthogonal to supporting free-threading.

You need to explicitly declare that matplotlib's extension modules don't use the GIL to mark matplotlib as thread-safe.

> What does "freethreading safe" mean?

Good question! The working definition we've been going with is that you should have similar thread safety guarantees under the GIL-enabled and GIL-disabled build. So any assumptions made in low-level code about the GIL providing locking and making it OK to do things like use C global statics to store module state are no longer OK and all that stuff should be fixed.

However there are lots of other thread safety bugs in Python codebases that are unrelated to the GIL, I don't think the presence of any of those issues means a module is "unsafe" under freethreading.

tacaswell commented:
with mpl 3.10.0 we are going to mark our selves as "thread safe", unless this is blocking something maybe wait for that to land on main before re-running this?

tacaswell commented:
or try this on the rc branch.

isuruf committed:
support python 3.14 freethreading

rm 3.13

isuruf committed:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.07.22.19.40.54

Other tools:
- conda-build 26.3.0
- rattler-build 0.63.1
- rattler-build-conda-compat 1.4.15
